src/confd/yang/infix-if-bridge@2024-03-06.yang

```diff
@@ -297,20 +314,42 @@ submodule infix-if-bridge {
           leaf-list untagged {
             type if:interface-ref;
             description "The set of ports in the untagged set for VLAN.";
+            must "/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name" {
+              error-message "Port not a member of this bridge.";
+            }
           }
 
           leaf-list tagged {
+            description "The set of ports in the tagged set for VLAN.";
             type if:interface-ref;
-            must "not(../untagged[contains(., current())])" {
-              error-message "Cannot be an untagged port at the same time.";
+            must "(/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name
+                   or current() = ../../../../if:name) and not(../untagged[contains(., current())])" {
+              error-message "Port not member of bridge or already set as untagged.";
```

Comment:
Can you have two separate `must` expressions on the same leaf? So that the user gets a more precise error message?

Reply:
Yes, all expressions separated by `and` can be placed in separate must expressions.  Good catch, fixing!

Reply:
Or no ... it works, technically, but breaks other functionality.

**Works:**
 - passes YANG validation with `yanglint`
 - gives better warnings for invalid configs

**Breaks:**
 - somehow confuses libyang/sysrepo so that
 - confd inference helpers stop working

For example, creating a VETH pair no longer works as we want.

**Conclusion:**

Could be a libyang bug, things start working again when I collapse two must expressions back into one.  So unfortunately the above expression has to remain as-is.